packages/delegation-controller/src/utils.ts

```diff
@@ -0,0 +1,79 @@
+import type { TypedMessageParams } from '@metamask/keyring-controller';
+import { getChecksumAddress } from '@metamask/utils';
+
+import { SIGNABLE_DELEGATION_TYPED_DATA } from './constants';
+import type { Address, Delegation, DelegationStruct } from './types';
+
+/**
+ * Checks if two addresses are equal.
+ *
+ * @param a - The first address.
+ * @param b - The second address.
+ * @returns True if the addresses are equal, false otherwise.
+ */
+export function isAddressEqual(a: Address, b: Address) {
```

Comment:
Minor, is it worth some unit tests for these functions, or do the existing tests cover all the branches?

Reply:
I believe current tests cover everything and that should be good enough. We could add tests for some of these, but writing tests for a one-liner seems overwhelming to me.